src/main/java/org/mitre/synthea/helpers/TransitionMetrics.java

```diff
@@ -105,138 +87,56 @@ public Metric getMetric(String moduleName, String stateName) {
-      if (!m.getClass().equals(Module.class)) {
-        // java module, not GMF. no states to show
-        continue;
-      }
-      System.out.println(m.name.toUpperCase());
-
-      Map<String, Metric> moduleMetrics = metrics.row(m.name);
-      List<String> keys = new ArrayList<String>(moduleMetrics.keySet());
-      Collections.sort(keys);
+  public static void exportMetrics() {
+    GsonBuilder builder = new GsonBuilder();
+    if (Config.getAsBoolean("exporter.pretty_print", true)) {
+      builder.setPrettyPrinting();
+    }
+    Gson gson = builder.create();
 
-      for (String stateName : keys) {
-        Metric stats = getMetric(m.name, stateName);
-        int entered = stats.entered.get();
-        int population = stats.population.get();
-        long duration = stats.duration.get();
-        int current = stats.current.get();
+    System.out.println("Saving metrics for " + metrics.rowKeySet().size() + " modules.");
 
-        System.out.println(stateName + ":");
-        System.out.println(" Total times entered: " + stats.entered);
-        System.out.println(" Population that ever hit this state: " + stats.population + " ("
-            + decimal(population, totalPopulation) + "%)");
-        System.out.println(" Average # of hits per total population: "
-            + decimal(entered, totalPopulation));
-        System.out.println(" Average # of hits per person that ever hit state: "
-            + decimal(entered, population));
-        System.out.println(" Population currently in state: " + stats.current + " ("
-            + decimal(current, totalPopulation) + "%)");
-        State state = m.getState(stateName);
-        if (state instanceof State.Guard || state instanceof State.Delay) {
-          System.out.println(" Total duration: " + durationOf(duration));
-          System.out.println(" Average duration per time entered: "
-              + durationOf(duration / entered));
-          System.out.println(" Average duration per person that ever entered state: "
-              + durationOf(duration / population));
-        } else if (state instanceof State.Encounter && ((State.Encounter) state).isWellness()) {
-          System.out.println(" (duration metrics for wellness encounter omitted)");
-        }
+    String baseDir = Config.get("exporter.baseDirectory", "./output/");
+    String statsDir = "metrics";
+    Path output = Paths.get(baseDir, statsDir);
+    output.toFile().mkdirs();
 
-        if (!stats.destinations.isEmpty()) {
-          System.out.println(" Transitioned to:");
-          long total = stats.destinations.values().stream().mapToLong(ai -> ai.longValue()).sum();
-          stats.destinations.forEach((toState, count) ->
-                System.out.println(" --> " + toState + " : " + count + " = "
-                                    + decimal(count.get(), total) + "%"));
-        }
-        System.out.println();
+    List<ModuleSupplier> suppliers = Module.getModuleSuppliers();
```

Comment:
Just a thought but if you modified `List<ModuleSupplier> getModuleSuppliers(Predicate<String> predicate)` in Module.java to take a Predicate<ModuleSupplier> instead then you could get rid of the loop that follows while also making that method more flexible.

Reply:
That sounds like a good idea. I'll take a look at it.

Reply:
OK, I made a change. Everything should be ready to go.